File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-doc-5-4/TradeIdentifier.cpp
// TradeIdentifier.cpp 
#include "TradeIdentifier.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

TradeIdentifier::TradeIdentifier(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //issuerNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* issuerNode = xmlNode->FirstChildElement("issuer");

   if(issuerNode){issuerIsNull_ = false;}
   else{issuerIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- issuerNode , address : " << issuerNode << std::endl;
   #endif
   if(issuerNode)
   {
      if(issuerNode->Attribute("href") || issuerNode->Attribute("id"))
      {
          if(issuerNode->Attribute("id"))
          {
             issuerIDRef_ = issuerNode->Attribute("id");
             issuer_ = boost::shared_ptr<IssuerId>(new IssuerId(issuerNode));
             issuer_->setID(issuerIDRef_);
             IDManager::instance().setID(issuerIDRef_,issuer_);
          }
          else if(issuerNode->Attribute("href")) { issuerIDRef_ = issuerNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { issuer_ = boost::shared_ptr<IssuerId>(new IssuerId(issuerNode));}
   }

   //tradeIdNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* tradeIdNode = xmlNode->FirstChildElement("tradeId");

   if(tradeIdNode){tradeIdIsNull_ = false;}
   else{tradeIdIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- tradeIdNode , address : " << tradeIdNode << std::endl;
   #endif
   if(tradeIdNode)
   {
      if(tradeIdNode->Attribute("href") || tradeIdNode->Attribute("id"))
      {
          if(tradeIdNode->Attribute("id"))
          {
             tradeIdIDRef_ = tradeIdNode->Attribute("id");
             tradeId_ = boost::shared_ptr<TradeId>(new TradeId(tradeIdNode));
             tradeId_->setID(tradeIdIDRef_);
             IDManager::instance().setID(tradeIdIDRef_,tradeId_);
          }
          else if(tradeIdNode->Attribute("href")) { tradeIdIDRef_ = tradeIdNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { tradeId_ = boost::shared_ptr<TradeId>(new TradeId(tradeIdNode));}
   }

   //partyReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* partyReferenceNode = xmlNode->FirstChildElement("partyReference");

   if(partyReferenceNode){partyReferenceIsNull_ = false;}
   else{partyReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partyReferenceNode , address : " << partyReferenceNode << std::endl;
   #endif
   if(partyReferenceNode)
   {
      if(partyReferenceNode->Attribute("href") || partyReferenceNode->Attribute("id"))
      {
          if(partyReferenceNode->Attribute("id"))
          {
             partyReferenceIDRef_ = partyReferenceNode->Attribute("id");
             partyReference_ = boost::shared_ptr<PartyReference>(new PartyReference(partyReferenceNode));
             partyReference_->setID(partyReferenceIDRef_);
             IDManager::instance().setID(partyReferenceIDRef_,partyReference_);
          }
          else if(partyReferenceNode->Attribute("href")) { partyReferenceIDRef_ = partyReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { partyReference_ = boost::shared_ptr<PartyReference>(new PartyReference(partyReferenceNode));}
   }

   //accountReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* accountReferenceNode = xmlNode->FirstChildElement("accountReference");

   if(accountReferenceNode){accountReferenceIsNull_ = false;}
   else{accountReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- accountReferenceNode , address : " << accountReferenceNode << std::endl;
   #endif
   if(accountReferenceNode)
   {
      if(accountReferenceNode->Attribute("href") || accountReferenceNode->Attribute("id"))
      {
          if(accountReferenceNode->Attribute("id"))
          {
             accountReferenceIDRef_ = accountReferenceNode->Attribute("id");
             accountReference_ = boost::shared_ptr<AccountReference>(new AccountReference(accountReferenceNode));
             accountReference_->setID(accountReferenceIDRef_);
             IDManager::instance().setID(accountReferenceIDRef_,accountReference_);
          }
          else if(accountReferenceNode->Attribute("href")) { accountReferenceIDRef_ = accountReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { accountReference_ = boost::shared_ptr<AccountReference>(new AccountReference(accountReferenceNode));}
   }

   //versionedTradeIdNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* versionedTradeIdNode = xmlNode->FirstChildElement("versionedTradeId");

   if(versionedTradeIdNode){versionedTradeIdIsNull_ = false;}
   else{versionedTradeIdIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- versionedTradeIdNode , address : " << versionedTradeIdNode << std::endl;
   #endif
   if(versionedTradeIdNode)
   {
      if(versionedTradeIdNode->Attribute("href") || versionedTradeIdNode->Attribute("id"))
      {
          if(versionedTradeIdNode->Attribute("id"))
          {
             versionedTradeIdIDRef_ = versionedTradeIdNode->Attribute("id");
             versionedTradeId_ = boost::shared_ptr<VersionedTradeId>(new VersionedTradeId(versionedTradeIdNode));
             versionedTradeId_->setID(versionedTradeIdIDRef_);
             IDManager::instance().setID(versionedTradeIdIDRef_,versionedTradeId_);
          }
          else if(versionedTradeIdNode->Attribute("href")) { versionedTradeIdIDRef_ = versionedTradeIdNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { versionedTradeId_ = boost::shared_ptr<VersionedTradeId>(new VersionedTradeId(versionedTradeIdNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<IssuerId> TradeIdentifier::getIssuer()
{
   if(!this->issuerIsNull_){
        if(issuerIDRef_ != ""){
             return boost::shared_static_cast<IssuerId>(IDManager::instance().getID(issuerIDRef_));
        }else{
             return this->issuer_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<IssuerId>();
   }
}
boost::shared_ptr<TradeId> TradeIdentifier::getTradeId()
{
   if(!this->tradeIdIsNull_){
        if(tradeIdIDRef_ != ""){
             return boost::shared_static_cast<TradeId>(IDManager::instance().getID(tradeIdIDRef_));
        }else{
             return this->tradeId_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<TradeId>();
   }
}
boost::shared_ptr<PartyReference> TradeIdentifier::getPartyReference()
{
   if(!this->partyReferenceIsNull_){
        if(partyReferenceIDRef_ != ""){
             return boost::shared_static_cast<PartyReference>(IDManager::instance().getID(partyReferenceIDRef_));
        }else{
             return this->partyReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PartyReference>();
   }
}
boost::shared_ptr<AccountReference> TradeIdentifier::getAccountReference()
{
   if(!this->accountReferenceIsNull_){
        if(accountReferenceIDRef_ != ""){
             return boost::shared_static_cast<AccountReference>(IDManager::instance().getID(accountReferenceIDRef_));
        }else{
             return this->accountReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AccountReference>();
   }
}
boost::shared_ptr<VersionedTradeId> TradeIdentifier::getVersionedTradeId()
{
   if(!this->versionedTradeIdIsNull_){
        if(versionedTradeIdIDRef_ != ""){
             return boost::shared_static_cast<VersionedTradeId>(IDManager::instance().getID(versionedTradeIdIDRef_));
        }else{
             return this->versionedTradeId_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<VersionedTradeId>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-doc-5-4/TradeIdentifier.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class TradeIdentifier : ISerialized
    {
        public TradeIdentifier(XmlNode xmlNode)
        {
            XmlNode issuerNode = xmlNode.SelectSingleNode("issuer");
            
            if (issuerNode != null)
            {
                if (issuerNode.Attributes["href"] != null || issuerNode.Attributes["id"] != null) 
                {
                    if (issuerNode.Attributes["id"] != null) 
                    {
                        issuerIDRef_ = issuerNode.Attributes["id"].Value;
                        IssuerId ob = new IssuerId(issuerNode);
                        IDManager.SetID(issuerIDRef_, ob);
                    }
                    else if (issuerNode.Attributes["href"] != null)
                    {
                        issuerIDRef_ = issuerNode.Attributes["href"].Value;
                    }
                    else
                    {
                        issuer_ = new IssuerId(issuerNode);
                    }
                }
                else
                {
                    issuer_ = new IssuerId(issuerNode);
                }
            }
            
        
            XmlNode tradeIdNode = xmlNode.SelectSingleNode("tradeId");
            
            if (tradeIdNode != null)
            {
                if (tradeIdNode.Attributes["href"] != null || tradeIdNode.Attributes["id"] != null) 
                {
                    if (tradeIdNode.Attributes["id"] != null) 
                    {
                        tradeIdIDRef_ = tradeIdNode.Attributes["id"].Value;
                        TradeId ob = new TradeId(tradeIdNode);
                        IDManager.SetID(tradeIdIDRef_, ob);
                    }
                    else if (tradeIdNode.Attributes["href"] != null)
                    {
                        tradeIdIDRef_ = tradeIdNode.Attributes["href"].Value;
                    }
                    else
                    {
                        tradeId_ = new TradeId(tradeIdNode);
                    }
                }
                else
                {
                    tradeId_ = new TradeId(tradeIdNode);
                }
            }
            
        
            XmlNode partyReferenceNode = xmlNode.SelectSingleNode("partyReference");
            
            if (partyReferenceNode != null)
            {
                if (partyReferenceNode.Attributes["href"] != null || partyReferenceNode.Attributes["id"] != null) 
                {
                    if (partyReferenceNode.Attributes["id"] != null) 
                    {
                        partyReferenceIDRef_ = partyReferenceNode.Attributes["id"].Value;
                        PartyReference ob = new PartyReference(partyReferenceNode);
                        IDManager.SetID(partyReferenceIDRef_, ob);
                    }
                    else if (partyReferenceNode.Attributes["href"] != null)
                    {
                        partyReferenceIDRef_ = partyReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        partyReference_ = new PartyReference(partyReferenceNode);
                    }
                }
                else
                {
                    partyReference_ = new PartyReference(partyReferenceNode);
                }
            }
            
        
            XmlNode accountReferenceNode = xmlNode.SelectSingleNode("accountReference");
            
            if (accountReferenceNode != null)
            {
                if (accountReferenceNode.Attributes["href"] != null || accountReferenceNode.Attributes["id"] != null) 
                {
                    if (accountReferenceNode.Attributes["id"] != null) 
                    {
                        accountReferenceIDRef_ = accountReferenceNode.Attributes["id"].Value;
                        AccountReference ob = new AccountReference(accountReferenceNode);
                        IDManager.SetID(accountReferenceIDRef_, ob);
                    }
                    else if (accountReferenceNode.Attributes["href"] != null)
                    {
                        accountReferenceIDRef_ = accountReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        accountReference_ = new AccountReference(accountReferenceNode);
                    }
                }
                else
                {
                    accountReference_ = new AccountReference(accountReferenceNode);
                }
            }
            
        
            XmlNode versionedTradeIdNode = xmlNode.SelectSingleNode("versionedTradeId");
            
            if (versionedTradeIdNode != null)
            {
                if (versionedTradeIdNode.Attributes["href"] != null || versionedTradeIdNode.Attributes["id"] != null) 
                {
                    if (versionedTradeIdNode.Attributes["id"] != null) 
                    {
                        versionedTradeIdIDRef_ = versionedTradeIdNode.Attributes["id"].Value;
                        VersionedTradeId ob = new VersionedTradeId(versionedTradeIdNode);
                        IDManager.SetID(versionedTradeIdIDRef_, ob);
                    }
                    else if (versionedTradeIdNode.Attributes["href"] != null)
                    {
                        versionedTradeIdIDRef_ = versionedTradeIdNode.Attributes["href"].Value;
                    }
                    else
                    {
                        versionedTradeId_ = new VersionedTradeId(versionedTradeIdNode);
                    }
                }
                else
                {
                    versionedTradeId_ = new VersionedTradeId(versionedTradeIdNode);
                }
            }
            
        
        }
        
    
        #region Issuer_
        private IssuerId issuer_;
        public IssuerId Issuer_
        {
            get
            {
                if (this.issuer_ != null)
                {
                    return this.issuer_; 
                }
                else if (this.issuerIDRef_ != null)
                {
                    issuer_ = IDManager.getID(issuerIDRef_) as IssuerId;
                    return this.issuer_; 
                }
                else
                {
                      return this.issuer_; 
                }
            }
            set
            {
                if (this.issuer_ != value)
                {
                    this.issuer_ = value;
                }
            }
        }
        #endregion
        
        public string issuerIDRef_ { get; set; }
        #region TradeId_
        private TradeId tradeId_;
        public TradeId TradeId_
        {
            get
            {
                if (this.tradeId_ != null)
                {
                    return this.tradeId_; 
                }
                else if (this.tradeIdIDRef_ != null)
                {
                    tradeId_ = IDManager.getID(tradeIdIDRef_) as TradeId;
                    return this.tradeId_; 
                }
                else
                {
                      return this.tradeId_; 
                }
            }
            set
            {
                if (this.tradeId_ != value)
                {
                    this.tradeId_ = value;
                }
            }
        }
        #endregion
        
        public string tradeIdIDRef_ { get; set; }
        #region PartyReference_
        private PartyReference partyReference_;
        public PartyReference PartyReference_
        {
            get
            {
                if (this.partyReference_ != null)
                {
                    return this.partyReference_; 
                }
                else if (this.partyReferenceIDRef_ != null)
                {
                    partyReference_ = IDManager.getID(partyReferenceIDRef_) as PartyReference;
                    return this.partyReference_; 
                }
                else
                {
                      return this.partyReference_; 
                }
            }
            set
            {
                if (this.partyReference_ != value)
                {
                    this.partyReference_ = value;
                }
            }
        }
        #endregion
        
        public string partyReferenceIDRef_ { get; set; }
        #region AccountReference_
        private AccountReference accountReference_;
        public AccountReference AccountReference_
        {
            get
            {
                if (this.accountReference_ != null)
                {
                    return this.accountReference_; 
                }
                else if (this.accountReferenceIDRef_ != null)
                {
                    accountReference_ = IDManager.getID(accountReferenceIDRef_) as AccountReference;
                    return this.accountReference_; 
                }
                else
                {
                      return this.accountReference_; 
                }
            }
            set
            {
                if (this.accountReference_ != value)
                {
                    this.accountReference_ = value;
                }
            }
        }
        #endregion
        
        public string accountReferenceIDRef_ { get; set; }
        #region VersionedTradeId_
        private VersionedTradeId versionedTradeId_;
        public VersionedTradeId VersionedTradeId_
        {
            get
            {
                if (this.versionedTradeId_ != null)
                {
                    return this.versionedTradeId_; 
                }
                else if (this.versionedTradeIdIDRef_ != null)
                {
                    versionedTradeId_ = IDManager.getID(versionedTradeIdIDRef_) as VersionedTradeId;
                    return this.versionedTradeId_; 
                }
                else
                {
                      return this.versionedTradeId_; 
                }
            }
            set
            {
                if (this.versionedTradeId_ != value)
                {
                    this.versionedTradeId_ = value;
                }
            }
        }
        #endregion
        
        public string versionedTradeIdIDRef_ { get; set; }
        public string choiceStr_0;
        public string choiceStr_1;
        
    
        
    
    }
    
}

